tests/test_http_downloader.py

```diff
@@ -0,0 +1,61 @@
+import tempfile
+from unittest.mock import patch
+
+import pytest
+
+from datasetinsights.io.downloader.http_downloader import HTTPDownloader
+from datasetinsights.io.exceptions import ChecksumError
+
+
+def test_read_checksum_from_txt():
+    with tempfile.NamedTemporaryFile(mode="w+") as tmp:
+        tmp.write("123456")
+        tmp.read()
+        assert HTTPDownloader.get_checksum_from_file(tmp.name) == 123456
+
+
+@patch("datasetinsights.io.downloader.http_downloader.download_file")
+@patch.object(HTTPDownloader, "read_checksum_from_txt")
+def test_get_checksum_from_file(
+    mock_read_checksum_from_txt, mock_download_file
+):
+    filepath_1 = "http://some/path"
+    filepath_2 = "https://some/path"
+    filepath_3 = "some/wrong/path"
+
+    mock_read_checksum_from_txt.return_value = 123456
```

Comment:
This defeats the purpose of the test. Mocking the function you are testing does not validate the behavior it just tests if mocking a function works.